### ISSUE TITLE
Unable to recreate a shared folder once it's deleted

### ISSUE THREAD

**Issue by vsajip** [lastpass/lastpass-cli#229]
Console session:

```
$ lpass status
Logged in as support@orbitil.com.
$ lpass share create FooBarBaz
$ lpass share userls FooBarBaz
Error: Share FooBarBaz not found.
$ lpass share userls Shared-FooBarBaz
User                                         RO  Admin   Hide OutEnt Accept
support@orbitil.com                           _      x      _      _      x
$ lpass share rm Shared-FooBarBaz
$ lpass share create FooBarBaz
$ lpass share userls Shared-FooBarBaz
Error: Share Shared-FooBarBaz not found.
```
There are a number of issues:

1. It's unintuitive and a bit clunky that you can't use the name you used to create the folder in the first place - you have to stick 'Shared-' in front whenever you refer to it. Is there a reason for this?
2. The second `lpass share create` gives no errors, but the second `lpass share userls` suggests that the shared folder wasn't recreated.
3. Not strictly a CLI issue, but in addition to the above, there seems to be no way of completely deleting shared folders, even in the Admin Console - they show up as e.g. `Shared-FooBarBaz (deleted)` and can't be completely removed, nor, it seems, can they be undeleted.

Is there a solution or workaround?

**Comment by bcopeland** [comment]
Going to answer these out of order since 2 depends on 3.

1. The main clunkiness is from 'create' accepting "FooBarBaz" in the first place.  When retrieving a shared folder, it always has to start with Shared-.  In the create case, it seems rude to abort completely if the naming scheme is not followed, but perhaps we should alert the user that "Shared-FooBarBaz" was created instead of "FooBarBaz".  I'm not sure whether we should make `lpass share userls Foo` just work -- we do know it should be Shared-Foo, but on the other hand, a bunch of other commands, like `lpass ls` will not know that you are interested in only shared folders and will fail, so doing that propagates the inconsistency around.

3. When you delete a shared folder, it goes into a deleted-but-not-purged state for, I think, 30 days (this allows recovery for accidental shares).  You should be able to purge them through the web interface.

2. ACK, I think this is just exiting with success since the folder is in this limbo state.

So, I propose the following changes:
 - for "lpass share create Foo", respond that folder "Shared-Foo" was created
 - for all other "lpass share" commands, if Foo is not found, but Shared-Foo exists in blob, say "share Foo not found, did you mean Shared-Foo?"
 - add `lpass share rm --purge`
 - make `lpass share add` for a deleted folder either display an error if there is an existing deleted folder, or undelete the existing folder

**Comment by bcopeland** [comment]
github markdown messed with my numbering; second answer is for point 3, and third answer is for point 2.

**Comment by vsajip** [comment]
> So, I propose the following changes

Yes, they seem reasonable and should fix this issue. Any idea of a timeline for getting this in?

**Comment by bcopeland** [comment]
I have some time next week to tackle these, so I'll be targeting the next release.

**Comment by wknapik** [comment]
\*bump\*

**Comment by primozcankar** [comment]
Any news on this. I have the same issue created a new folder, shared it as an empty folder then deleted it and now I can't recreate it as a shared folder.

**Comment by bcopeland** [comment]
I wound up implementing the first of those bullet points but not the rest.  @primozcankar there is a simple work-around today though: go into the vault's Sharing Center, and click 'Show deleted shared folders', and then click on the trash can to purge.

**Comment by MeOnGitHub123** [comment]
seems not working (anymore) ?!

```
$lpass share rm --purge
#lpass: invalid option -- p
#lpass: invalid option -- u
#Usage: lpass share rm SHARE

$lpass share rm <SharedFolder> 
#Error: Share <SharedFolder> not found.
```

to bad not being able to purge deleted folders via CLI 
much appreciated when you find time 

Am I correct there is no way  (in the GUI)  to purge multiple entries (want to get rid of all my testing shared-folders)
![CLI_purgefails](https://user-images.githubusercontent.com/13984479/56907063-cfcc5880-6aa3-11e9-9fb9-17fae54aa07d.jpg)
  ?
